impl/src/metadata.rs

```diff
@@ -54,128 +41,11 @@ pub struct CargoWorkspaceFiles {
- * https://github.com/rust-lang/cargo/blob/0.40.0/src/cargo/core/package.rs#L32-L50)
- */
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
-pub struct Package {
-  pub name: String,
-  pub version: String,
-  pub id: PackageId,
-  pub license: Option<String>,
-  pub license_file: Option<String>,
-  pub description: Option<String>,
-  pub source: Option<String>,
-  pub dependencies: Vec<Dependency>,
-  pub targets: Vec<Target>,
-  pub features: HashMap<String, Vec<String>>,
-  pub manifest_path: String,
-  pub edition: String,
-  pub sha256: Option<String>,
-}
-
-/**
- * The metadata for a dependency (a reference connecting a crate to another crate).
- *
- * WARNING: Cargo-raze does not control the definition of this struct.
- * This struct mirrors Cargo's own [`SerializedDependency`](
- * https://github.com/rust-lang/cargo/blob/0.40.0/src/cargo/core/dependency.rs#L49-L60)
- */
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
-pub struct Dependency {
-  pub name: String,
-  pub source: String,
-  pub req: String,
-  pub kind: Option<Kind>,
-  #[serde(default = "default_dependency_field_optional")]
-  pub optional: bool,
-  #[serde(default = "default_dependency_field_uses_default_features")]
-  pub uses_default_features: bool,
-  pub features: Vec<String>,
-  pub target: Option<TargetSpec>,
-}
-
-/**
- * The metadata for a compileable target.
- *
- * WARNING: Cargo-raze does not control the definition of this struct.
- * This struct mirrors Cargo's own [`SerializedTarget`](
- * https://github.com/rust-lang/cargo/blob/0.40.0/src/cargo/core/manifest.rs#L188-L197)
- */
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
-pub struct Target {
-  pub name: String,
-  pub kind: Vec<String>,
-  pub crate_types: Vec<String>,
-  pub src_path: String,
-  pub edition: String,
-}
-
-/**
- * The metadata for a fully resolved dependency tree.
- *
- * WARNING: Cargo-raze does not control the definition of this struct.
- * This struct mirrors Cargo's own [`MetadataResolve`](
- * https://github.com/rust-lang/cargo/blob/0.40.0/src/cargo/ops/cargo_output_metadata.rs#L91-L95)
- */
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
-pub struct Resolve {
-  pub nodes: Vec<ResolveNode>,
-  pub root: PackageId,
-}
-
-/**
- * The metadata for a single resolved entry in the full dependency tree.
- *
- * WARNING: Cargo-raze does not control the definition of this struct.
- * This struct mirrors Cargo's own [`Node`](
- * https://github.com/rust-lang/cargo/blob/0.40.0/src/cargo/ops/cargo_output_metadata.rs#L102-L106)
- */
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
-pub struct ResolveNode {
-  pub id: PackageId,
-  pub dependencies: Vec<PackageId>,
-  // Optional due to recent feature addition in Cargo.
-  pub features: Option<Vec<String>>,
-}
-
 /** A workspace metadata fetcher that uses the Cargo Metadata subcommand. */
 pub struct CargoSubcommandMetadataFetcher {
```

Comment:
Nit, maybe this deserves a name change? 

Not really sure what to call it though. Maybe just CargoMetadataFetcher? That doesn't really distinguish the type very well, but I don't like the next best alternative (CargoCrateMetadataFetcher) either because "Crate" is ambiguous there.

No strong request here unless you have a suggestion (or you're OK with either just CargoMeta(..) or CargoCrateMeta(...).